spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/CachingConnectionFactory.java

```diff
@@ -1295,4 +1316,23 @@ else if (!RabbitUtils.isNormalChannelClose(cause)) {
 
 	}
 
+
+    /**
+     * Name each create thread by this class.
+     */
+    private static class AmqpNamedThreadPoolFactory implements ThreadFactory {
+        private AtomicInteger threadId = new AtomicInteger();
+
+        @Override
+        public Thread newThread(Runnable r) {
+            Thread t = new Thread(r);
+            t.setDaemon(true);
```

Comment:
Why does it become `daemon`?
Does looks like a `DefaultThreadFactory` does the same:
```
    if (t.isDaemon())
                t.setDaemon(false);
```
Well, it does exactly opposite.

On the other hand I wonder why `CustomizableThreadFactory` doesn't work here for us?..